browser/ui/webui/brave_wallet/ledger/ledger_ui.cc

```diff
@@ -0,0 +1,53 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/webui/brave_wallet/ledger/ledger_ui.h"
+
+#include <string>
+
+#include "brave/components/constants/webui_url_constants.h"
+#include "brave/components/ledger_bridge/resources/grit/ledger_bridge_generated_map.h"
+#include "components/grit/brave_components_resources.h"
+#include "content/public/browser/web_contents.h"
+#include "content/public/browser/web_ui_data_source.h"
+#include "ui/resources/grit/webui_generated_resources.h"
+
+namespace ledger {
+
+UntrustedLedgerUI::UntrustedLedgerUI(content::WebUI* web_ui)
+    : ui::UntrustedWebUIController(web_ui) {
+  auto* untrusted_source =
+      content::WebUIDataSource::Create(kUntrustedLedgerURL);
+  untrusted_source->SetDefaultResource(IDR_BRAVE_WALLET_LEDGER_BRIDGE_HTML);
+  untrusted_source->AddResourcePaths(
+      base::make_span(kLedgerBridgeGenerated, kLedgerBridgeGeneratedSize));
+  untrusted_source->AddFrameAncestor(GURL(kBraveUIWalletPageURL));
+  untrusted_source->AddFrameAncestor(GURL(kBraveUIWalletPanelURL));
+  untrusted_source->OverrideContentSecurityPolicy(
+      network::mojom::CSPDirectiveName::ScriptSrc,
+      std::string("script-src chrome://resources/js/ 'self';"));
```

Comment:
What JS resources are we linking from here that makes us need access to this?

Reply:
Hmm, I got this code from the Trezor [implementation](https://github.com/brave/brave-core/blob/783525e2804b1c3bcfa4bdc74db9de23db924654/browser/ui/webui/brave_wallet/trezor/trezor_ui.cc#L32-L35), which may have different requirements since I think it actually uses third party scripts.  If this line isn't needed to run the scripts in ledger.ts (added in this pull request), this can be removed.

Reply:
I'm not exactly certain either, but I think if it's not necessary we should remove. Can you try removing and see if it breaks anything?

cc @diracdeltas to see if she may know more about this

Reply:
i'm not sure either, would just try removing it and seeing if anything breaks.

Reply:
Seems to work without it, so I removed it in https://github.com/brave/brave-core/pull/14096/commits/fe99f17f7f3b02e9e19c874f7f3a00fad4c2e92c

Reply:
Thanks for checking, resolving this one.